File: CookMaster/Models/NutritionInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CookMaster.Models
{
    public class NutritionInfo
    {
        [PK]
        public int RecipeID { get; set; }

        public string Calories { get; set; }
        public string Carbs { get; set; }
        public string Fat { get; set; }
        public string Protein { get; set; }

        [NotStoredInDB]
        public List<BadGoodItem> Bad { get; set; }

        [NotStoredInDB]
        public List<BadGoodItem> Good { get; set; }

        [NotStoredInDB]
        public List<Nutrient> Nutrients { get; set; }

        [NotStoredInDB]
        public List<Property> Properties { get; set; }

        [NotStoredInDB]
        public List<Flavonoid> Flavonoids { get; set; }

        [NotStoredInDB]
        public List<Ingredient> Ingredients { get; set; }

        [NotStoredInDB]
        public CaloricBreakdown CaloricBreakdown { get; set; }

        [NotStoredInDB]
        public WeightPerServing WeightPerServing { get; set; }

        [NotStoredInDB]
        public long Expires { get; set; }

        [NotStoredInDB]
        public bool Stale { get; set; }
    }
}

File: CookMaster.WebApp/ClientApp/src/app/shared/models/nutrition-info.ts
import {BadGoodItem} from './bad-good-item';
import {Nutrient} from './nutrient';
import {Property} from './property';
import {Flavonoid} from './flavonoid';
import {Ingredient} from './ingredient';
import {CaloricBreakdown} from './caloric-breakdown';
import {WeightPerServing} from './weight-per-serving';

export interface NutritionInfo {
  RecipeID: number;

  Calories: string;
  Carbs: string;
  Fat: string;
  Protein: string;

  Bad: BadGoodItem[];
  Good: BadGoodItem[];
  Nutrients: Nutrient[];
  Properties: Property[];
  Flavonoids: Flavonoid[];
  Ingredients: Ingredient[];

  CaloricBreakdown: CaloricBreakdown;
  WeightPerServing: WeightPerServing;

  Expires: number;
  Stale: boolean;
}
